### PR TITLE
[SoXResampler] link to OpenMP

### PR DESCRIPTION
**Pull request by bramtayl**

### CODE REVIEW

File: S/SoXResampler/build_tarballs.jl [hunk 1]
```diff
@@ -3,19 +3,18 @@
...
-        "b797a5d23078be234e520af1041b5e11b49864696d56f0d0b022a0349d1e8d1b"
-    )
+    ArchiveSource("https://pilotfiber.dl.sourceforge.net/project/soxr/soxr-$(version)-Source.tar.xz", "b111c15fdc8c029989330ff559184198c161100a59312f5dc19ddeb9b5a15889")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/soxr-code-*
+cd $WORKSPACE/srcdir
+apk add libgomp
```
Comment (giordano) [inline S/SoXResampler/build_tarballs.jl:16]: Libogmp is a runtime dependency of the target...
```suggestion
```

File: S/SoXResampler/build_tarballs.jl [hunk 2]
```diff
@@ -33,7 +32,7 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
+    Dependency(PackageSpec(name="OpenMPI_jll", uuid="fe0851c0-eecd-5654-98d4-656369965a5c"))
```
Comment (giordano) [inline S/SoXResampler/build_tarballs.jl:35]: Why did you remove `CompilerSupportLibraries_jll` if you want OpenMP?  Also, how is OpenMPI related?

Comment (bramtayl) [reply S/SoXResampler/build_tarballs.jl:35]: Well, removing compiler support libraries was an accident...I think libgomp and OpenMPI are different implementations of OpenMP? I added both to hedge my bet. I'm still not sure how to figure out whether something is a run-time or a compile-time dependency. The build reports:

```
[17:41:46] -- Found OpenMP_C: -fopenmp (found version "3.1") 
[17:41:46] -- Found OpenMP: TRUE (found version "3.1")
```

So one of them must have worked?

Comment (giordano) [reply S/SoXResampler/build_tarballs.jl:35]: > I think libgomp and OpenMPI are different implementations of OpenMP?

No, libgomp is an implementation of [OpenMP](https://en.wikipedia.org/wiki/OpenMP), OpenMPI is an implementation of [MPI](https://en.wikipedia.org/wiki/Message_Passing_Interface)

Comment (giordano) [reply S/SoXResampler/build_tarballs.jl:35]: > So one of them must have worked?

Target libgomp is _always_ there during the build, `CompilerSupportLibraries_jll` is more important for runtime

Comment (bramtayl) [reply S/SoXResampler/build_tarballs.jl:35]: It's already there? I tried Pkg.develop ing and I'm getting new errors in my in-progress PulseAudio build:

```
/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/4.8.5/../../../../x86_64-linux-gnu/bin/ld: warning: libgomp.so.1, needed by /workspace/destdir/lib/libsoxr.so.0, not found (try using -rpath or -rpath-link)
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `omp_get_num_threads@OMP_1.0'
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `GOMP_parallel_start@GOMP_1.0'
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `omp_set_lock@OMP_3.0'
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `omp_destroy_lock@OMP_3.0'
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `omp_get_thread_num@OMP_1.0'
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `omp_init_lock@OMP_3.0'
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `GOMP_parallel_end@GOMP_1.0'
/workspace/destdir/lib/libsoxr.so.0: undefined reference to `omp_unset_lock@OMP_3.0'
```

Comment (giordano) [reply S/SoXResampler/build_tarballs.jl:35]: > `try using -rpath or -rpath-link`

Comment (giordano) [reply S/SoXResampler/build_tarballs.jl:35]: It's really easier for me if we're on the same page and you push whatever changes you have locally.

Comment (bramtayl) [reply S/SoXResampler/build_tarballs.jl:35]: Sure! Sorry I'm so hopeless at this. Current try is at #2469

**Comment by bramtayl** [comment]
Just an update that even if I disable soxr I get a similar error with unrelated packages not finding each other. I'm not sure exactly how to use rpath? I've tried adding `rpath=/usr/lib/` and `rpath-link=/usr/lib` to `c_args` and `c_link_args` and neither seemed to help...I wonder if on master `ld` isn't looking in the right places?

**Comment by giordano** [comment]
You don't like providing context, do you? :slightly_smiling_face:

Give as much information as you can (ideally a *[minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)*) and someone will maybe have a look.  If you don't give any clue of what you're doing and how you're getting "errors", we can do little to help you

**Comment by bramtayl** [comment]
Right, did you see #2469? It's not exactly minimal but it will get you there. Run as is to get the problem with libgomp, and add the `-Dsoxr=disabled` flag to get

```
/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/4.8.5/../../../../x86_64-linux-gnu/bin/ld: warning: libz.so.1, needed by /workspace/destdir/lib/libgio-2.0.so, not found (try using -rpath or -rpath-link)
```

**Comment by bramtayl** [comment]
Note also that `apk del` doesn't seem to be working:


```
sandbox:${WORKSPACE}/srcdir/pulseaudio-14.2/build # apk del gettext
/sbin/apk: unrecognized option: no-chown
```

**Comment by giordano** [comment]
> Right, did you see #2469?

Frankly?  No.  Pulseaudio is one of those package which will take months to get it right, I personally don't have time to dedicate to a similar effort for at least a couple of months nor the littlest interest in it.  I actually just opened the script, and see that CI doesn't even start, so nothing for me to see.  However I noticed that you install some 40 (forty!) packages for the host, which I believe I told you already a few times won't help you at all to build libraries for the target.

**Comment by bramtayl** [comment]
Ok, I understand about PulseAudio being a total pain. I installed all the packages I could as both jlls and via apk just to cover all my bases. If I run into the same `ld` issue with a smaller package I'll let you know.
